tests/posix/common/src/clock.c

```diff
@@ -70,8 +71,31 @@ void test_posix_realtime(void)
 	ret = clock_gettime(CLOCK_REALTIME, &rts);
 	zassert_equal(ret, 0, "Fail to get realtime clock");
 
-	zassert_equal(rts.tv_sec, mts.tv_sec, "Seconds not equal");
-	zassert_equal(rts.tv_nsec, mts.tv_nsec, "Nanoseconds not equal");
+	/* Between both clock_gettime(), time may have been increased by 1ms
```

Comment:
It's nice reminder that the time keeping has granularity, besides precision. I.e., the fact that somewhere time is kept in nanoseconds in milliseconds, doesn't mean it increments in such units. But in that regard, I don't think we can guarantee a particular granularity across Zephyr architectures and versions. For example, I remember seeing granularity of 10ms. So, I'd suggest to remove references to "1ms" and use more generic wording, and then apparently also introduce a symbolic constant for granularity (and be ready to bump it up, though it can start with 1ms).